build.rs

```diff
@@ -390,6 +397,10 @@ fn find_boost_include_dir() -> Result<PathBuf, Box<dyn Error>> {
     .into())
 }
 
+fn contains_subslice<T: PartialEq>(haystack: &[T], needle: &[T]) -> bool {
+    haystack.windows(needle.len()).any(|w| w == needle)
```

Comment:
Wouldn't this match if *any* of "-D" or "warnings" match, rather than both (consecutively)? Or false positives for anything with "-D" in the middle?

This seems unlikely to hit false positives in RUSTFLAGS, but it'd be really confusing if it did, so I want to double check.

Reply:
`windows(needle.len())` returns a sequence of subslices, and `w == needle` tests the slices for equality which should mean checking they contain the exact same elements in the same order

Reply:
[windows() in playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0def39d4a8115f3cbbd540b4ed3c6862)

Reply:
Okay -- `needle.len()` refers to consecutive _words_ here rather than the character length of the individual needle elements, so false positives from substring matches aren't an issue.